### ISSUE TITLE
Getting an error

### ISSUE THREAD

**Issue by RoySRose** [graalvm/simpletool#4]
Hello, First of all, thank you for the project.
I'm trying to create a simple `helloworld` program of instrumenting js through Graal VM.
Eventually to create a monitoring tool for Graal.

But having a hard time. Since there isn't any instructions on how to execute(with what options) in the [instrumenting guide in graal homepage](https://www.graalvm.org/docs/graalvm-as-a-platform/implement-instrument/#simple-instrumentation-agent)

I've found this repository and try to execute one. So far I've understood or assumed... that I need to compile the instrumenting program which uses truffle API(in this case SimpleCoverageInstrument.java and all) and gives the jar as an option when running js program (Please let me know if I am getting this wrong)

I've maven installed and got `simpletool-19.2.0.jar`
and then I've executed `runJSWithCoverage.sh` and is giving me an argument error.

![image](https://user-images.githubusercontent.com/20402016/69026580-cf2eeb80-0a0e-11ea-9d94-bbcf53364d48.png)

Am I doing something wrong? Can you point out a few things if I'm on the wrong path?

![image](https://user-images.githubusercontent.com/20402016/69026727-672cd500-0a0f-11ea-8ab7-7c30c178c0af.png)

**Comment by boris-spas** [comment]
Hey @RoySRose,
Thanks for taking the time to look into GraalVM instrumentation!
Yeah, the documentation for instrumentation is currently a work in progress (as is this repo which is meant to be a starting point for development), hopefully we will have some better guides soon :) 

So, I just double checked that the setup here works for me. What I did was point `JAVA_HOME` to `GRAALVM_HOME` and run
`mvn clean package && ./runJsWithCoverage.sh`
Seems like you are on a mac and I am running linux so maybe (though not likely) there is a subtle difference of some kind... 

Maybe we can do a few sanity checks:
- Is the `simpletool-19.2.0.jar` in the `target` folder?
- could you provide the output of `./simpletool js --version:graalvm` 
- check that `./simpletool js --help:tools` contains a `Simple Code Coverage` entry

**Comment by RoySRose** [comment]
@boris-spas 
Hello, Thank you for your time taking look at this. As you've commented I also don't think it's the mac that causes this. Well, it's what Graalvm is for, 'running multiple languages on one vm'. (of course, there is also native part)
I've followed your instructions and checked as below

1. I've changed $JAVA_HOME to $GRAALVM_HOME in `simpletool` script
 ![image](https://user-images.githubusercontent.com/20402016/69107810-fa204a80-0ab5-11ea-9111-a482bcf9d944.png)

2.  my $GRAALVM_HOME points to `graalvm-ce-19.2.1`
 ![image](https://user-images.githubusercontent.com/20402016/69107821-02788580-0ab6-11ea-85ad-714a8357a717.png)

3. Yes there is `simpletool-19.2.0.jar` in the `target` folder?
 ![image](https://user-images.githubusercontent.com/20402016/69107916-55ead380-0ab6-11ea-9448-eb4f5f64ee4a.png)
(just added few `print` in `example.js`)

4. I can't run `./simpletool js --version:graalvm` with below error. 
 ![image](https://user-images.githubusercontent.com/20402016/69107953-7c107380-0ab6-11ea-8a50-2fb279672a8d.png)

But I can run it if I remove ` --simple-code-coverage \` from `simpletool` script, and the result is as below
 ![image](https://user-images.githubusercontent.com/20402016/69108004-a7935e00-0ab6-11ea-8fa1-7ac8ee814bc3.png)

5. I'm trying to check `./simpletool js --help:tools` contains a `Simple Code Coverage` entry, I've run the command but I'm afraid I don't know how to check the entry. posting the results below. If you can guide me more on this. I will check right away.

 ![image](https://user-images.githubusercontent.com/20402016/69108391-d65e0400-0ab7-11ea-890e-f8c7a5b6e057.png)

Please let me know if there is anything for me to check additionally.

**Comment by boris-spas** [comment]
Unusual... Seems that the tool is not picked up when being appended to the classpath. 
Would you please run the following to make sure it's not a path issue.
`$GRAALVM_HOME/bin/js --jvm --vm.Dtruffle.class.path.append=/absolute/path/to/simpletool-19.2.0.jar --simple-code-coverage -e '1+1'`
for me the output is as follows:
```
==
Coverage of null is 100.00%
+ 1+1
```

**Comment by RoySRose** [comment]
Unfortunately... no luck

![image](https://user-images.githubusercontent.com/20402016/69203112-dd077c80-0b86-11ea-8b48-d158bb9d7d0f.png)

seems `--simple-code-coverage` option(?)/tool(?) is still not recognized.

So far I know, Graalvm CE 19.2 is the latest version. Do you think I should try with another version?

-------------------------

Have tried with graalvm-ce-19.1.1, still no luck..

![image](https://user-images.githubusercontent.com/20402016/69216300-38e5fb80-0baf-11ea-8895-310062ed5142.png)

any suggestions? that I can try?

**Comment by boris-spas** [comment]
As a sanity check I re-ran the gate with 19.2.0 (https://travis-ci.org/boris-spas/simpletool/jobs/614411116?utm_medium=notification&utm_source=github_status) and with 19.2.1 (https://travis-ci.org/boris-spas/simpletool/jobs/614414957?utm_medium=notification&utm_source=github_status) and both pass. 

I'm running low on ideas about whats special about your machine :/

**Comment by RoySRose** [comment]
@boris-spas 

First of all, Once again thank you for your time.
I looked at your travis link.
Decided to start from scratch. 

Fortunately, It worked. 

Got back and searched for the reason. 
I found out the option was not being registered.
```text
@Option(name = "", help = "Enable Simple Coverage (default: fa......
```
Something wrong about auto generated `SimpleCoverageInstrumentOptionDescriptors.class`

But it was due to some kind of dependency error. I couldn't dig it deeper. But it was solved after re-updating the dependencies.

Anyway. Thank you again.
I'll start looking into truffle instrument api, hope to reach you with good new.
